File: MatchaLatte.Web.Angular/src/app/shared/pagination-result.ts
export class PaginationResult<T> {
  pageIndex: number;
  pageSize: number;
  itemCount: number;
  items: T[] = [];

  constructor(pageIndex?: number, pageSize?: number, itemCount?: number, items?: T[]) {
    this.pageIndex = pageIndex || 0;
    this.pageSize = pageSize || 10;
    this.itemCount = itemCount || 0;
    this.items = items || [];
  }
}

File: src/MatchaLatte.Common/Queries/PaginationResult.cs
using System.Collections.Generic;

namespace MatchaLatte.Common.Queries
{
    /// <summary>
    /// 分頁結果。
    /// </summary>
    /// <typeparam name="TResult">結果類型。</typeparam>
    public class PaginationResult<TResult>
    {
        public ICollection<TResult> Items { get; set; }

        public int ItemCount { get; set; }
    }
}